### PR TITLE
plater: init at 2020-07-30

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
I need this for my 3D printer.

While they do have a 1.0 release that was before they got their build system
working well and fixed some major bugs, so I opted to package the last commit,
which I have tested to work.

cc. @flokli

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -22989,6 +22989,8 @@ in
 
   pistol = callPackage ../tools/misc/pistol { };
 
+  plater = callPackage ../applications/misc/plater { };
```
Comment (doronbehar) [inline pkgs/top-level/all-packages.nix:22992]:
```suggestion
  plater = libsForQt5.callPackage ../applications/misc/plater { };
```

File: pkgs/applications/misc/plater/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Rhoban";
+    repo = "Plater";
+    rev = "f8de6d038f95a9edebfcfe142c8e9783697d5b47";
+    sha256 = "0r20mbzd16zv1aiadjqdy7z6sp09rr6lgfxhvir4ll3cpakkynr4";
+  };
+
+  nativeBuildInputs = [ cmake qt5.wrapQtAppsHook ];
+  buildInputs = [ libGLU qt5.qtbase ];
```
Comment (doronbehar) [inline pkgs/applications/misc/plater/default.nix:19]:
```suggestion
  buildInputs = [ libGLU qtbase ];
```

File: pkgs/applications/misc/plater/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  version = "2020-07-30";
+
+  src = fetchFromGitHub {
+    owner = "Rhoban";
+    repo = "Plater";
+    rev = "f8de6d038f95a9edebfcfe142c8e9783697d5b47";
+    sha256 = "0r20mbzd16zv1aiadjqdy7z6sp09rr6lgfxhvir4ll3cpakkynr4";
+  };
+
+  nativeBuildInputs = [ cmake qt5.wrapQtAppsHook ];
```
Comment (doronbehar) [inline pkgs/applications/misc/plater/default.nix:18]: Use `mkDerivation` from libsForQt5

```suggestion
  nativeBuildInputs = [ cmake ];
```

File: pkgs/applications/misc/plater/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{ stdenv
+, cmake
+, fetchFromGitHub
+, lib
+, libGLU
+, qt5
+}: stdenv.mkDerivation rec {
```
Comment (doronbehar) [inline pkgs/applications/misc/plater/default.nix:7]:
```suggestion
}:

mkDerivation rec {
```

File: pkgs/applications/misc/plater/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{ stdenv
```
Comment (doronbehar) [inline pkgs/applications/misc/plater/default.nix:1]:
```suggestion
{ mkDerivation
```

File: pkgs/applications/misc/plater/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+, cmake
+, fetchFromGitHub
+, lib
+, libGLU
+, qt5
```
Comment (doronbehar) [inline pkgs/applications/misc/plater/default.nix:6]:
```suggestion
, qtbase
```

**Review by doronbehar — CHANGES_REQUESTED** [review]

**Comment by lovesegfault** [comment]
Made changes, thanks for the review @doronbehar!

**Review by doronbehar — APPROVED** [review]
Seems good to me, and I'll merge unless you'd like to wait for flokli's approval.

**Comment by flokli** [comment]
There's a trailing whitespace at meta.description, other than that, LGTM. Didn't test building or running it, but I assume you did ;-)

**Comment by lovesegfault** [comment]
@flokli good catch! Fixed!

**Pull request merged**